### PR TITLE
Adds a new LithoHandler implementation that uses Executors

### PR DESCRIPTION
**Pull request by chggr**
## Summary

This change adds the ExecutorLithoHandler, which is a LithoHandler
implementation that runs all layout computations against a given
Executor. The ExecutorLithoHandler can be used in apps that already
have a well established threading model and would like to use one
of their pre-existing Executors to run all layout against.

## Changelog

Adds a new LithoHandler implementation that uses Executors.

## Test Plan

Unit tests have been added as part of the change.

### CODE REVIEW

File: litho-core/build.gradle [hunk 1]
```diff
@@ -70,6 +70,7 @@ dependencies {
 
     // Android Support Library
     compileOnly deps.supportAnnotations
+    implementation deps.guava
```
Comment (yschimke) [inline litho-core/build.gradle:73]: Are there any non-guava alternatives here? Multiset is useful but guava can be a lightning rod dependency. Or is FB Android code ok with Guava?

Comment (chggr) [reply litho-core/build.gradle:73]: There are other places within Litho that are currently using guava, check litho/lib/guava/BUCK

Comment (yschimke) [reply litho-core/build.gradle:73]: I got the impression it was only test code, n.p. if it's already used then.

**Comment by chggr** [comment]
Hi @adityasharat, I hope you are well!

Just wondering whether you have any thoughts or feedback on this PR. Thanks!

**Comment by adityasharat** [comment]
From @astreet 
> I don't think we want to depend on guava in litho-core (unless that's changed?). Even if it has, this change seems like a not great reason to start pulling it in. Maybe this should just go in another module, like a utils one? Or change the impl to not depend on guava and use another datastructure impl.

File: litho-core/build.gradle (hunk 1 above)
Comment (astreet) [reply litho-core/build.gradle:73]: It's only used in the annotation processor and test code so it doesn't get shipped in any apps using Litho.

Comment (chggr) [reply litho-core/build.gradle:73]: I see, thanks for the context. I will remove usages of MultiSet from the ExecutorLithoHandler, and also the dependency on guava.

**Comment by chggr** [comment]
Many thanks for the feedback! I have now changed the implementation to not depend on Guava at all, as recommended.

**Pull request closed without merging**